agent/engine/ordering_integ_test.go

```diff
@@ -339,6 +339,72 @@ func TestDependencyHealthyTimeout(t *testing.T) {
 	waitFinished(t, finished, orderingTimeout)
 }
 
+
+func TestGranularStopTimeout(t *testing.T) {
+	taskEngine, done, _ := setupWithDefaultConfig(t)
+	defer done()
+
+	stateChangeEvents := taskEngine.StateChangeEvents()
+
+	taskArn := "testDependencyHealthyTimeout"
+	testTask := createTestTask(taskArn)
+
+	dependency := createTestContainerWithImageAndName(baseImageForOS, "dependency")
+	parent1 := createTestContainerWithImageAndName(baseImageForOS, "parent1")
+	parent2 := createTestContainerWithImageAndName(baseImageForOS, "parent2")
+
+	dependency.EntryPoint = &entryPointForOS
+	dependency.Command = []string{"sleep 30"}
+	dependency.Essential = true
+
+	parent1.EntryPoint = &entryPointForOS
+	parent1.Command = []string{"trap 'echo caught' SIGTERM; while true; do echo hello; sleep 1s; done;"}
+	parent1.DependsOn = []apicontainer.DependsOn{
+		{
+			Container: "dependency",
+			Condition: "START",
+		},
+	}
+	parent1.StopTimeout = 5
+
+	parent2.EntryPoint = &entryPointForOS
+	parent2.Command = []string{"trap 'echo caught' SIGTERM; sleep 60"}
+	parent2.DependsOn = []apicontainer.DependsOn{
+		{
+			Container: "dependency",
+			Condition: "START",
+		},
+	}
+	parent2.StopTimeout = 50
+
+	testTask.Containers = []*apicontainer.Container{
+		parent1,
+		parent2,
+		dependency,
+	}
+
+	go taskEngine.AddTask(testTask)
+
+	finished := make(chan interface{})
+	go func() {
+
+		verifyContainerRunningStateChange(t, taskEngine)
+		verifyContainerRunningStateChange(t, taskEngine)
+		verifyContainerRunningStateChange(t, taskEngine)
+
+		verifyTaskIsRunning(stateChangeEvents, testTask)
+
+		verifyContainerStoppedStateChange(t, taskEngine)
+		verifyContainerStoppedStateChange(t, taskEngine)
+		verifyContainerStoppedStateChange(t, taskEngine)
```

Comment:
This test will succeed regardless of if the timeouts are applied or not. We need to validate the exit codes of the containers.